niceplots/utils.py

```diff
@@ -1,127 +1,120 @@
+import contextlib
 import matplotlib.pyplot as plt
 import numpy as np
-from cycler import cycler
-from collections import OrderedDict
 from .parula import parula_map
-from matplotlib import patheffects
+from matplotlib import patheffects, rc_context, style, rcdefaults
 from matplotlib.collections import LineCollection
 import matplotlib.colors as mcolor
 import warnings
+import os
 
 
-def setRCParams(dark_mode=False, set_dark_background=False):
+def setStyle(styleName="doumont-light", afterReset=False):
     """
-    Set some defaults for generating nice, Doumont-esque plots.
+    Set some defaults for generating nice plots.
 
     Parameters
     ----------
-    dark_mode (optional) : bool
-        If true, sets axes, labels, etc. to white so the plot
-        can be used on a dark background.
-        NOTE: Unless you are explicitly saving the plot with
-        a transparent background (e.g. as a png with
-        transparent=True), also set the
-        set_dark_background option.
-    set_dark_background (optional) : bool or str
-        If true, sets the axis and figure backgrounds to black.
-        This option can also be set to a color string, such as
-        "#aaaaaa", to use a color other than black.
+    styleName : str, optional
+        Name of desired style. Avaiable styles are:
+
+            - doumont-light: the niceplots style you know and love
+            - doumont-dark: the dark version of the niceplots style you know and love
+            - james-dark: a really cool alternative to classic niceplots
+            - james-light: a version of james with a light background, naturally
+
+    afterReset : bool, optional
+        If True, will first reset the rcParams to the matplotlib then apply
+        the new rcParams. The alternative (when afterReset is False) is to apply
+        the new rcParams on top of the current rcParams.
+
+    Returns
+    -------
+    dict
+        Dictionary of the colors for the requested style. The keys
+        are human-readable names and the keys are the hex codes.
     """
```

Comment:
I am personally not a fan of functions returning things that their names don't suggest they do. Feel like it would be better to not return this here and just have users call `getColors` when they want to get the colors

Reply:
Yeah that's a good point.